Add Interaction command

MagnificentCreature opened the pull request:
Fixes issue #7 and #4.
Adds Interaction as a field under person, a few dependencies would have to be modified but has not been changed yet.
This does not do UI changes
Some test cases been created to check if interaction is working.
Also note that Interaction is stored as a set (of interactions) in person. This could lead to considerations like
1)How do edit/remove interactions
2)How to handle the fact that interactions is a set

MagnificentCreature committed:
First commit

MagnificentCreature committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2103T-F11-4/tp

MagnificentCreature committed:
Added vscode settings and change .gitignore

MagnificentCreature committed:
change gitignore

MagnificentCreature committed:
WIP: Add Interaction class & interaction command

lilozz2 commented on .vscode/settings.json:
Should we put this in .gitignore?

MagnificentCreature replied:
Will do!

MagnificentCreature committed:
Changed gitignore as per @lilozz2 suggestion

MagnificentCreature committed:
WIP

MagnificentCreature committed:
Change Checkstyle for LineLength, interaction WIP
Also change gitignore again as per @lilozz2 suggestion, the change earlier didn't seem to have done it

MagnificentCreature committed:
Remove settings.json

MagnificentCreature committed:
Implement InteractionCommandParser
But have not tested functionality yet
Changed ParserUtil to include parseInteraction
Changed InteractionCommand to include interaction

MagnificentCreature committed:
WIP Interaction command

MagnificentCreature committed:
Fix checkstyle issue

MagnificentCreature committed:
Implement core set of logic for interaction

MagnificentCreature committed:
Bug fixes for interaction command

MagnificentCreature committed:
Bug fixes
Fix the issue of running "interaction 1 I am interested in this product" which will throw a formatting error because
the preamble is read as "1 I am interested in this product" and the index cannot be parsed.

lilozz2 commented on config/checkstyle/checkstyle.xml:
This was for suppressing the checkstyle warnings right

MagnificentCreature replied:
Yes, in particular for a line that was 122 length long, I found a workaround by moving it onto the next line, so technically, this can be removed. But I think its also okay to keep, it doesn't affect the code in any way but might be useful in the future when we want to stop the checkstyle from pinging long lines (ie regex)

lilozz2 commented on src/main/java/seedu/address/logic/parser/CreateCommandParser.java:
```diff
@@ -68,8 +70,11 @@ public CreateCommand parse(String args) throws ParseException {
+        // command does not allow adding interactions right away
 
-        Person person = new Person(name, phone, email, address, tagList, telegram, profession, income, details);
+        Person person = new Person(name, phone, email, address, tagList, telegram, profession, income, details,
+                interaction);
```
```suggestion
        Person person = new Person(name, phone, email, address, tagList, telegram, profession, income, details)
```

lilozz2 replied:
I think the constructor should not have the interaction

MagnificentCreature replied:
This is useful for when we are copying data onto CSV and loading from CSV files later?

Let me know if you think this is valid, else I'll remove it then

MagnificentCreature replied:
Okay, removed!

KamiliArsyad commented on src/main/java/seedu/address/model/person/Person.java:
```diff
@@ -30,6 +30,7 @@ public class Person {
     private final Profession profession;
     private final Income income;
     private final Details details;
+    private Set<Interaction> interactions = new HashSet<>();
```
can be final, too coz the set object itself will not be altered

MagnificentCreature replied:
Will change!

KamiliArsyad reviewed (changes requested):
I think other than the interaction in `Person` constructor that needs to be removed, everything else is good. As a side note, generally bloating the constructor is not gud and I'd planned to group all the properties of a person inside one single abstraction (e.g. sth like `PersonProperty`) but hasn't got the chance to do that yet.

MagnificentCreature committed:
Added the change @KamiliArsyad suggested

dickongwd reviewed (commented):
Should interaction still be in the constructor for `Person`? We can consider adding an interaction to a `Person` through an `addInteraction` interface for example, and initialising `interaction` as null.

MagnificentCreature committed:
Add tests and fixes by @lilozz2 @KamiliArsyad
